### ISSUE TITLE
With auto diff base, Percy will use any PR branch that happens to be named `master` as the baseline

### ISSUE THREAD

**Issue by rsimha-amp** [percy/percy-client#26]
We're seeing this on `ampproject/amphtml`, which uses forked branches for development. If a developer inadvertently pushes a PR from their local `master` branch, and if that build contains visual diffs, those snapshots are used as the baseline for all subsequent Percy builds until a new commit is merged to the real `master`.

/cc @fotinakis @timhaines

**Comment by rsimha-amp** [comment]
See https://percy.io/ampproject/amphtml

[Build # 5314](https://percy.io/ampproject/amphtml/builds/364343) is actually a PR build from https://github.com/ampproject/amphtml/pull/11566, but it was pushed from a developer's local `master` branch, and ended up being used by Percy as the baseline for subsequent builds.

**Comment by fotinakis** [comment]
Yeah this is a hard one, Percy doesn't really assume a difference between you running something locally or in CI (ie. we don't actually _require_ that you're running something in CI for it to set a state in Percy). The docs do lightly mention this:

> Careful though—if you run your tests locally, Percy cannot read the current branch information from the CI environment so it will default to master which will set your repo's master baseline in Percy. You can change the branch linked in Percy by setting the PERCY_BRANCH environment variable.
https://percy.io/docs/setup/local#local-setup

If you know deterministically for AMP that you're not in CI, you could set something like `PERCY_BRANCH=local-development` to override it, but changing it globally would require a bit more thought on our end of what the behavior for us should be across libraries and environments.

**Comment by rsimha-amp** [comment]
Right, I hear you that it's difficult to detect and deal with this across all your libraries.

If I do use the `PERCY_BRANCH=local-development` workaround, would it mean that _all_ Percy build will start showing that as the branch name, instead of the names selected by different users? (That'd be bad, since it's useful for committers to see their real branch names on the Percy dashboard.)

I'll give this some thought too. Thanks!

**Comment by rsimha-amp** [comment]
Also, to clarify, the Percy build in the original report was indeed generated from Travis CI, and not from a developer's local machine. It just happened to be pushed from the `master` branch of the developer's local fork.

**Comment by rsimha-amp** [comment]
@fotinakis Thanks for the tip on `PERCY_BRANCH`. I've mitigated this at our end via https://github.com/ampproject/amphtml/pull/11579. 

However, the original issue still remains on the Percy side, where it will use any PR build from a branch named `master` as the new baseline, whether it originates from CI or from a local attempt. Of course, this can only happen when development is done from forked repos (like we do), and the build originates from `master` on the forked repo.

Feel free to close this if you think it's a corner case not worth fixing.
